microovn/ovn/cmd/commands.go

```diff
@@ -86,20 +134,42 @@ func NewOvsdbSpec(dbType OvsdbType) (*OvsdbSpec, error) {
 	return dbSpec, err
 }
 
+// certArgs returns list of arguments that can be directly used when executing OVN/OVS commands
+// that require connection over SSL. These arguments configure paths to client certificate, client
+// private key and CA certificate.
+func certArgs() []string {
+	cert, key := paths.PkiClientCertFiles()
+	return []string{
+		"-C", paths.PkiCaCertFile(),
+		"-c", cert,
+		"-p", key,
+	}
+}
+
 // WaitForDBState as the name suggests, waits for specified ovsdb database to settle in
 // specified state. If database does not reach this state within timeout, this function returns error.
-// SocketURL specified in "db" parameter does not need to necessarily exist before this function is executed,
-// creation of the database socket (db.SocketURL) will be awaited as well.
+// If SocketURL specified in "db" parameter points to a local unix socket, it does not need to exist before
+// this function is executed, creation of the socket  file will be awaited as well.
 func WaitForDBState(ctx context.Context, _ state.State, db *OvsdbSpec, dbState string, timeout int) error {
+	args := []string{
+		"--timeout",
+		strconv.Itoa(timeout),
+	}
+
+	socketURL := db.SocketURL
+	// Insert certificates arguments if DB connection uses SSL
+	if strings.HasPrefix(db.SocketURL, "ssl") {
+		args = append(args, certArgs()...)
+		// ovsdb-client is not capable to connect to clustered database,
+		// so we select just one of the SSL endpoints
+		socketURL, _, _ = strings.Cut(socketURL, ",")
+	}
+	args = append(args, "wait", socketURL, db.Name, dbState)
+
```

Comment:
Yes, this function suffers a bit because it's used by many commands in this package (nbctl/sbctl/vsctl/ovsdb-client), so it has to service many scenarios. I'm not happy with the ugly workaround using first IP either, I'll investigate how to improve this.

Reply:
Now that the `NBCtlCluster` uses `ovn-nbctl` wrapper, this function went back to being used only for local unix sockets.